lang/semgrep-grammars/src/semgrep-promql/grammar.js

```diff
@@ -39,6 +39,31 @@ module.exports = grammar(base_grammar, {
 
     function_args: ($) =>
       seq("(", commaSep(choice($.semgrep_ellipsis, $._query)), ")"),
+
+    binary_expression: ($) => {
```

Comment:
Did you try instead to allow '...' for _query instead?
So that we don't need special treatment for binary_expression?

Reply:
Can we go with the special treatment for now? Allowing `...` for _query in general leads to some conflicts that i cannot yet grasp the implications of.

Reply:
ok. Maybe you could add a TODO that it would be better to generalize and your current findings about why we get conflicts
by doing it the clean way.

Reply:
The problem I have with your current solution is that it's copy pasting code from the original grammar in table, and so if one modify the original grammar to add an operator, we will not get a warning about it.
Maybe we could export table in the original grammar so that it could simply be referenced here, instead of copy-pasted.

Reply:
I understand; do you mind if i clean it up as followup? ill try writing the grammar friendlier to semgrep now that i have a better understanding hwo it works.